### PR TITLE
Remove unused icons

### LINKED ISSUE #10557: Delete unused icons
While working on #10536, I removed some icons that weren't used anymore. I think the `Icon` component has some icons that we are not using and we could delete the files as well as remove some of the CSS. If we don't need those icons, there is no need to keep them around, we can re-add them later.

While working on this issue, we should look at the other components to see if they embed files that aren't useful (we usually have an `img/` folder for images in the component's folder).

### PR DESCRIPTION
**Pull request by iamvaishnav**
Fixes #10557 

Delete icons that are no longer in use.

### CODE REVIEW

File: src/amo/components/Icon/styles.scss [hunk 1]
```diff
@@ -290,33 +249,10 @@
   @include permission-icon($name: 'navigation');
 }
 
-.Icon-developer {
-  @include icon($name: 'developer');
-}
```
Comment (willdurand) [inline src/amo/components/Icon/styles.scss:295]: We need this one in the user profile of a developer, e.g., https://addons-dev.allizom.org/en-US/firefox/user/10641750/

File: src/amo/components/Icon/styles.scss [hunk 2]
```diff
@@ -290,33 +249,10 @@
...
-
-.Icon-artist {
-  @include icon($name: 'artist');
-}
```
Comment (willdurand) [inline src/amo/components/Icon/styles.scss:299]: We need this one in the user profile of a (theme) artist, e.g., https://addons-dev.allizom.org/en-US/firefox/user/8699116/

**Review by willdurand — CHANGES_REQUESTED** [review]

**Comment by willdurand** [comment]
This is a good start, thanks!

File: src/amo/components/Icon/styles.scss (hunk 1 above)
Comment (iamvaishnav) [reply src/amo/components/Icon/styles.scss:295]: I'm sorry. I will revert it.

**Review by willdurand — APPROVED** [review]
Looks good (hopefully). Thanks!

**Comment by willdurand** [comment]
Thanks @iamvaishnav for your first contribution! /cc @caitmuenster

**Comment by caitmuenster** [comment]
Thank you so much for the patch, @iamvaishnav! Your contribution has been added to our [recognition wik](https://wiki.mozilla.org/Add-ons/Contribute/Recognition). 

Welcome onboard! We hope to see you around the project. 😎

**Pull request merged**